server/jwt/jwt.go

```diff
@@ -0,0 +1,162 @@
+package jwt
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/fabiocicerchia/go-proxy-cache/config"
+	"github.com/lestrrat-go/backoff/v2"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/lestrrat-go/jwx/jwt"
+	log "github.com/sirupsen/logrus"
+)
+
+var co *config.Jwt
+var jwtKeyFetcher *jwk.AutoRefresh
+
+func InitJWT(conifg *config.Jwt) {
+
+	if co == nil {
+		co = conifg
+		jwtKeyFetcher = jwk.NewAutoRefresh(co.Context)
+		jwtKeyFetcher.Configure(co.Jwks_url,
+			jwk.WithMinRefreshInterval(15*time.Minute),
```

Comment:
the magic number `15` should either be defined as a constant or, even better, to be configurable.

Reply:
done!